File: vigenere.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(int argc, string argv[])
{
    //Check if correct number of arguments.
    if (argc != 2)
    {
        printf("Error: Wrong number of arguments.\n");
        return 1;
    }

    string keyword = argv[1];

    //Check if argument contains non-alphabetical character.
    for (int i = 0; i < strlen(keyword); i++)
    {
        keyword[i] = toupper(keyword[i]);
        if (keyword[i] < 'A' || keyword[i] > 'Z')
        {
            printf("Error: Keyword contains non-alphabetical character.\n");
            return 1;
        }
    }

    string text = get_string("plaintext: ");
    int j = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] >= 'A' && text[i] <= 'Z')
        {
            text[i] = (keyword[j] - 'A' + text[i] - 'A') % 26 + 'A';
            j++;
        }

        else if (text[i] >= 'a' && text[i] <= 'z')
        {
            text[i] = (keyword[j] - 'A' + text[i] - 'a') % 26 + 'a';
            j++;
        }
        j %= strlen(keyword);
    }
    printf("ciphertext: %s\n", text);
}

File: vigenere.py
from cs50 import get_string
import sys


def main():
    if len(sys.argv) != 2:
        print("Error: Wrong number of arguments.")
        sys.exit(1)
        return 1

    keyword = sys.argv[1].upper()

    if not keyword.isalpha():
        print("Error: Keyword contains non-alphabetical character.")
        sys.exit(1)
        return 1

    text = get_string("plaintext: ")
    ciphertext = ""
    j = 0

    for char in text:

        k_val = ord(keyword[j]) - ord("A")

        if char.isupper():
            char = (ord(char) - ord("A") + k_val) % 26 + ord("A")
            j += 1
        elif char.islower():
            char = (ord(char) - ord("a") + k_val) % 26 + ord("a")
            j += 1
        else:
            char = ord(char)

        ciphertext += chr(char)
        j %= len(keyword)

    print(u"ciphertext: {}".format(ciphertext))


if __name__ == "__main__":
    main()